src/EventStore.Core/Services/PersistentSubscription/PersistentSubscriptionService.cs

```diff
@@ -171,29 +171,31 @@ public void Handle(ClientMessage.CreatePersistentSubscription message)
                 return;
             }
 
-            CreateSubscriptionGroup(message.EventStreamId, 
-                                    message.GroupName, 
-                                    message.ResolveLinkTos, 
+            var msgTimeout = message.MessageTimeoutMilliseconds == 0 ? TimeSpan.MaxValue : TimeSpan.FromMilliseconds(message.MessageTimeoutMilliseconds);
+            var checkPointAfter = message.CheckPointAfterMilliseconds == 0 ? TimeSpan.MaxValue : TimeSpan.FromMilliseconds(message.CheckPointAfterMilliseconds);
+            CreateSubscriptionGroup(message.EventStreamId,
+                                    message.GroupName,
+                                    message.ResolveLinkTos,
                                     message.StartFrom,
                                     message.RecordStatistics,
                                     message.MaxRetryCount,
                                     message.LiveBufferSize,
                                     message.BufferSize,
                                     message.ReadBatchSize,
-                                    TimeSpan.FromMilliseconds(message.CheckPointAfterMilliseconds),
```

Comment:
@gregoryyoung  Here and line 188 - times are reversed.  message timeout being used as checkpoint and visa versa.

Side note in the web ui the timespan is being displayed as `-2147483648` when using `DontTimeout` - not a big deal